Update build and integrate KVM

Toasterson opened the pull request:
These are my latest changes based from xen0l/oi-packer and currently the base of my Vagrantcloud boxes. The configuration got migrated to hcl2 which is new in packer  1.5 Additionally I integrated KVM support from PR #2 

Currently, the VirtualBox VM works and is available at: https://app.vagrantup.com/Toasterson/boxes/openindiana-hipster
The KVM build fails to bring up the NIC, due to the fact, that OpenIndiana identifies the NIC inside the Vagrant box as a different NIC to the one used during installation. Thus, no network configuration is present for the new NIC. I still need to find a solution for that issue but otherwise the box works.

Toasterson commented:
Currently latest releases are available at https://app.vagrantup.com/Toasterson/boxes/openindiana-hipster/versions/2020.04.13 will run both builds again next month with current release.

pyhalov commented on README.md:
````diff
@@ -18,24 +18,39 @@ Please note, that box building works only on platforms where packer runs.
-  packer build hipster.json
-  ```
+```bash
+git clone https://github.com/Toasterson/oi-packer.git && \
````
Likely this should refer OI repo

Toasterson replied:
Yes it should and does now :)

pyhalov commented:
Can you rebase it on master to avoid conflicts?

Toasterson committed:
Migrate to hcl2 configuration language
Update README accordingly
Add KVM support as seen in PR #2
Update box versions to 2020.05.04

Toasterson commented:
Rebased as asked does this help?

Toasterson commented:
KVM build is fixed BTW. both builds work and produce running boxes

Toasterson commented:
Needed to remove an email that accidentally ended up in my git config.
